compiler/rustc_const_eval/src/interpret/step.rs

```diff
@@ -320,6 +320,22 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                 let discr_val = self.read_discriminant(&op)?.0;
                 self.write_scalar(discr_val, &dest)?;
             }
+
+            OffsetOf(ref fields, container) => {
+                let container =
+                    self.subst_from_current_frame_and_normalize_erasing_regions(container)?;
+                let mut layout = self.layout_of(container)?;
+                let mut offset = 0;
+
+                for field in fields {
+                    offset += layout.fields.offset(field.index()).bytes();
+
+                    layout = layout.field(self, field.index());
+                }
```

Comment:
Would it make sense to factor this into a helper function that is shared with the codegen backends?

Reply:
I think it would, but I've got no idea where such a helper should be placed

Reply:
`rustc_const_eval::util` should be fine.

Reply:
I ended up putting it on `TyAndLayout` because the signature fit matched its methods more than anything in `rustc_const_eval::util`